### PR TITLE
Oglmodel

### PR DESCRIPTION
**Pull request by CorentinLanguepin**
Finalizing with changes mostly on scene and contactheader

### CODE REVIEW

**Comment by damienmarchal** [comment]
Hi @Younesssss and @CorentinLanguepin 

Thanks for the PR. 
I do not merge it directly as this is currently part of a larger process composed:
- adding to SofaPython3 the missing functions to smooth the step between sofapython & sofapython3
- split what was in STLIB in two libraries : 
     - SPLIB:
     - STLIB: 
- For both libraries, adding unittest so that can be integrated into the sofa CI stack.
- For each library have a:
    - SPLIB/splib and SPLIB/splib3, 
    - STLIB/stlib and STLIB/stlib3, 
splib/stlib will focus on implementing exactly the same feature as before but on top of SofaPython3 while the  
splib3/stlib3 version will be focused on providing new features or alternative design using the new special features of SofaPython3. 
The underlying idea of all that is to maximize the backward compatibility.

**Comment by EulalieCoevoet** [comment]
Hi Damien, 

As you said this is a large process and a work in progress. Meanwhie, some of us are using the branch `sofaPython3` as a temporary solution to have things working. 

This PR is actually to merge the work of Corentin into the branch `sofaPython3`, and I don't see why we souldn't. 
I'll start to review the changes if it's okay with you.

File: python3/src/stlib3/physics/rigid/RigidObject.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ def RigidObject(name="RigidObject",
                 volume=1.,
                 inertiaMatrix=[1., 0., 0., 0., 1., 0., 0., 0., 1.],
                 color=[1., 1., 0.],
+                collisionGroup='0',
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/physics/rigid/RigidObject.py]: I think the default behavior is "everything can collide", so 0 is not a good default value. I'll change that.

File: python3/src/stlib3/physics/rigid/RigidObject.py [hunk 2]
```diff
@@ -48,19 +49,22 @@ def RigidObject(name="RigidObject",
...
     object.addObject('UniformMass', name="mass", vertexMass=[totalMass, volume, inertiaMatrix[:]])
 
-    if not isAStaticObject:
-        plugins.append('SofaConstraint')
-        object.addObject('UncoupledConstraintCorrection')
+
+    #if not isAStaticObject:
+        #plugins.append('SofaConstraint')
+        #object.addObject('EulerImplicitSolver')
+        #object.addObject('CGLinearSolver')
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/physics/rigid/RigidObject.py]: I'm not sure why this has been removed? @CorentinLanguepin @Younesssss 
same for the `RequiredPlugin`.

File: python3/src/stlib3/physics/rigid/RigidObject.py [hunk 3]
```diff
@@ -111,5 +117,7 @@ def createScene(root):
     scene.addSimulation()
 
     ## Create a RigidObject with a cube mesh.
-    rigid = RigidObject(surfaceMeshFileName="mesh/smCube27.obj", parent=scene.Modelling)
-    scene.Simulation.addChild(rigid)
+    rigid = RigidObject(surfaceMeshFileName="mesh/smCube27.obj")
+    scene.Modelling.addChild(rigid)
+    print("HELLO")
+    #scene.Simulation.addChild(rigid)
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/physics/rigid/RigidObject.py]: A reminder to remove the print.

File: python3/src/stlib3/scene/contactheader.py [hunk 4]
```diff
@@ -1,9 +1,9 @@
 # -*- coding: utf-8 -*-
 
-def ContactHeader(applyTo, alarmDistance, contactDistance, frictionCoef=0.0):
+def ContactHeader(rootNode, alarmDistance, contactDistance, frictionCoef=0.0):
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/scene/contactheader.py]: I like `applyTo` better as it means something.

File: python3/src/stlib3/scene/contactheader.py [hunk 5]
```diff
@@ -25,24 +25,22 @@ def ContactHeader(applyTo, alarmDistance, contactDistance, frictionCoef=0.0):
...
-
-    if applyTo.getObject("FreeMotionAnimationLoop", warning=False) is None:
-            applyTo.addObject('FreeMotionAnimationLoop')
+                        angleCone=0.1)
+    rootNode.addObject('GenericConstraintSolver', tolerance=1e-5, maxIterations=5e2)
 
-    if applyTo.getObject("GenericConstraintSolver", warning=False) is None:
-            applyTo.addObject('GenericConstraintSolver', tolerance=1e-6, maxIterations=1000)
+    if rootNode.hasObject('SparseLDLSolver') is None:
+            rootNode.addObject('SparseLDLSolver')
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/scene/contactheader.py]: I think it still should be `GenericConstraintSolver` and not `SparseLDLSolver`, those are completely different solvers.

File: python3/src/stlib3/scene/scene.py [hunk 6]
```diff
@@ -59,29 +61,35 @@ def addMainHeader(plugins=plugins, repositoryPaths=repositoryPaths,doDebug=False
...
-                return root
+        def addDefaultSolver(node):
+            node.addObject('EulerImplicitSolver', name='TimeIntegrationSchema')
+            if iterative:
+                return node.addObject('CGLinearSolver', name='LinearSolver')
+            else:
+                node.addObject('SparseLDLSolver', name='LinearSolver')
+                node.addObject('MechanicalMatrixMapper', template='Vec3d,Rigid3d',
+                object1='@./RigidifiedStructure/DeformableParts/dofs', object2='@./RigidifiedStructure/RigidParts/dofs',
+                nodeToParse='@./RigidifiedStructure/DeformableParts/ElasticMaterialObject')
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/scene/scene.py]: This is not generic at all...

File: python3/src/stlib3/visuals/visualmodel.py [hunk 7]
```diff
@@ -44,3 +44,5 @@ def createScene(root):
     visu = VisualModel(visualMeshPath="mesh/smCube27.obj")
     visu.showGrid()
     scene.Modelling.addChild(visu)
+
+    #scene.Modelling.init()
```
Comment (EulalieCoevoet) [inline python3/src/stlib3/visuals/visualmodel.py]: to remove

**Review by EulalieCoevoet — CHANGES_REQUESTED** [review]

**Comment by EulalieCoevoet** [comment]
I've made the changes and tested them. It's all good for me.

**Pull request merged**